File: src/TurboVision/System/TClipboard.cs
using System.Runtime.InteropServices;

namespace TurboVision.System;

public static unsafe partial class TClipboard
{
    internal static partial class NativeMethods
    {
        [LibraryImport(Global.DLL_NAME, StringMarshalling = StringMarshalling.Utf8)]
        public static partial Error TV_TClipboard_setText(string text);

        [LibraryImport(Global.DLL_NAME)]
        public static partial Error TV_TClipboard_requestText();
    }
}

File: src/tvision4c/System/TClipboard.cpp
#include "TClipboard.h"

EXPORT tv::Error TV_TClipboard_setText(const char* text) {
    if (!text) {
        return tv::Error_ArgumentNull;
    }

    TClipboard::setText(text);
    return tv::Success;
}

EXPORT tv::Error TV_TClipboard_requestText() {
    TClipboard::requestText();
    return tv::Success;
}
